partition/grub_test.go

```diff
@@ -28,18 +28,25 @@ import (
 	. "gopkg.in/check.v1"
 
 	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
 )
 
 func grubEnvPath() string {
 	return filepath.Join(dirs.GlobalRootDir, "boot/grub/grubenv")
 }
 
 func grubEditenvSet(c *C, key, value string) {
+	if !osutil.FileExists("/usr/bin/grub-editenv") {
+		c.Skip("/usr/bin/grub-editenv is not available")
```

Comment:
This patch is invalid. All distributions I am aware of (except for Debian/Ubuntu) have the grub commands with `grub2-*` instead of `grub-*`. So the proper fix is to test for `grub2-*` variants of the commands and use those, otherwise fall back to `grub-*`.